praw/models/reddit/subreddit.py

```diff
@@ -1881,6 +1899,193 @@ def update(self, **settings):
         )
 
 
+class SubredditModerationStream:
+    """Provides moderator streams."""
+
+    def __init__(self, subreddit):
+        """Create a SubredditModerationStream instance.
+
+        :param subreddit: The moderated subreddit associated with the streams.
+
+        """
+        self.subreddit = subreddit
+
+    def edited(self, only=None, **stream_options):
+        """Yield edited comments and submissions as they become available.
+
+        :param only: If specified, one of ``'comments'``, or ``'submissions'``
+            to yield only results of that type.
+
+        Keyword arguments are passed to :func:`.stream_generator`.
+
+        For example, to retrieve all new edited submissions/commented made
+        to all moderated subreddits, try:
+
+        .. code:: python
+
+           for item in reddit.subreddit('mod').mod.stream.edited():
+               print(item)
+
+        """
+        return stream_generator(
+            self.subreddit.mod.edited, only=only, **stream_options
+        )
+
+    def log(self, action=None, mod=None, **stream_options):
+        """Yield moderator log entries as they become available.
+
+        :param action: If given, only return log entries for the specified
+            action.
+        :param mod: If given, only return log entries for actions made by the
+            passed in Redditor.
+
+        For example, to retrieve all new mod actions made to all moderated
+        subreddits, try:
+
+        .. code:: python
+
+           for log in reddit.subreddit('mod').mod.stream.log():
+               print("Mod: {}, Subreddit: {}".format(log.mod, log.subreddit))
+
+        """
+        return stream_generator(
+            self.subreddit.mod.log,
+            action=action,
+            mod=mod,
+            attribute_name="id",
+            **stream_options
+        )
+
+    def modmail_conversations(
+        self, other_subreddits=None, sort=None, state=None, **stream_options
+    ):
+        """Yield unread moderator messages as they become available.
+
+        :param other_subreddits: A list of :class:`.Subreddit` instances for
+            which to fetch conversations (default: None).
+        :param sort: Can be one of: mod, recent, unread, user
+            (default: recent).
+        :param state: Can be one of: all, archived, highlighted, inprogress,
+            mod, new, notifications, (default: all). "all" does not include
+            internal or archived conversations.
+
+        Keyword arguments are passed to :func:`.stream_generator`.
+
+        To print new mail in the unread modmail queue try:
+
+        .. code:: python
+
+           for message in reddit.subreddit('all').mod.stream.modmail_conversations():
+               print("From: {}, To: {}".format(message.author, message.dest))
```

Comment:
When I try this code sample I get an exception. Perhaps this is because of some misconfiguration on my end, and if so it would be nice to have a note explaining why this may happen. PRAW users are likely to try the example snippets to see how the library works, so hopefully we can avoid them getting exceptions or at least explain why it may happen.

Here's the full traceback I get, for debugging purposes:

```
In [8]: %paste                                                                  

           for message in reddit.subreddit('all').mod.stream.modmail_conversations():
               print("From: {}, To: {}".format(message.author, message.dest))

## -- End pasted text --
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-8-7b3d16fface0> in <module>
      1 for message in reddit.subreddit('all').mod.stream.modmail_conversations():
----> 2     print("From: {}, To: {}".format(message.author, message.dest))

~/PycharmProjects/jarhill0praw/praw/models/reddit/base.py in __getattr__(self, attribute)
     32         if not attribute.startswith("_") and not self._fetched:
     33             self._fetch()
---> 34             return getattr(self, attribute)
     35         raise AttributeError(
     36             "{!r} object has no attribute {!r}".format(

~/PycharmProjects/jarhill0praw/praw/models/reddit/base.py in __getattr__(self, attribute)
     35         raise AttributeError(
     36             "{!r} object has no attribute {!r}".format(
---> 37                 self.__class__.__name__, attribute
     38             )
     39         )

AttributeError: 'ModmailConversation' object has no attribute 'author'

```

Reply:
The attrs are wrong. Will be fixed in my next commit.

Reply:
Here's what I'm getting with the change I made in the docs:
```python
>>> for message in reddit.subreddit('all').mod.stream.modmail_conversations():
>>>     print("From: {}, To: {}".format(message.owner, message.participant))
>>>
From: PewdiepieSubmissions, To: Levi_Made_Me_Gay_77
From: MemeEconomy, To: DukePotato0620
From: PewdiepieSubmissions, To: weebmemer69420
From: PewdiepieSubmissions, To: WhatTheHeckWhoStoleM
From: PewdiepieSubmissions, To: EthanQa17
From: PewdiepieSubmissions, To: ItsYoHomie_
From: pics, To: beefaronii6
From: MemeEconomy, To: goblin_2748
From: MemeEconomy, To: Byby16
From: PewdiepieSubmissions, To: Bond_Backup
From: MurderedByWords, To: AutoModerator
From: pics, To: scottishswan
From: PewdiepieSubmissions, To: 69bigboypp69
From: MakingaMurderer, To: Mattyice002
From: MurderedByWords, To: AutoModerator
From: PewdiepieSubmissions, To: ZoutigeKip
From: MurderedByWords, To: AutoModerator
From: pics, To: Skankinzombie22
From: PewdiepieSubmissions, To: Blared_Unicorn
From: Justfuckmyshitup, To: mostlikelynotellie
From: dankmemes, To: FailURGamer24```

Reply:
Here's what I'm getting with the change I made in the docs:
```python 
>>> for message in reddit.subreddit('all').mod.stream.modmail_conversations():
>>>    print("From: {}, To: {}".format(message.owner, message.participant))
>>>
From: PewdiepieSubmissions, To: Levi_Made_Me_Gay_77
From: MemeEconomy, To: DukePotato0620
From: PewdiepieSubmissions, To: weebmemer69420
From: PewdiepieSubmissions, To: WhatTheHeckWhoStoleM
From: PewdiepieSubmissions, To: EthanQa17
From: PewdiepieSubmissions, To: ItsYoHomie_
From: pics, To: beefaronii6
From: MemeEconomy, To: goblin_2748
From: MemeEconomy, To: Byby16
From: PewdiepieSubmissions, To: Bond_Backup
From: MurderedByWords, To: AutoModerator
From: pics, To: scottishswan
From: PewdiepieSubmissions, To: 69bigboypp69
From: MakingaMurderer, To: Mattyice002
From: MurderedByWords, To: AutoModerator
From: PewdiepieSubmissions, To: ZoutigeKip
From: MurderedByWords, To: AutoModerator
From: pics, To: Skankinzombie22
From: PewdiepieSubmissions, To: Blared_Unicorn
From: Justfuckmyshitup, To: mostlikelynotellie
From: dankmemes, To: FailURGamer24```

Reply:
sorry for the multiple comments. I hit the wrong button and accidentally started a review.

Reply:
It is for new modmail. I based the documentation on the parent method here: https://github.com/praw-dev/praw/blob/0c198a2eb47e001cc4bec82353f45e826be96c2c/praw/models/reddit/subreddit.py#L2521

Reply:
Will the old modmail be supported?

Reply:
Yes, `unread` is old modmail. Sorry for commenting in the wrong stop lol. Page moved on me and I didn’t realize it.

Reply:
Ok thanks

Reply:
@LilSpazJoekp can we have a note in `unread` then attesting to that fact? Thanks!

Reply:
Absolutely I will get that committed really quick.